code/datums/antagonists/datum_traitor.dm

```diff
@@ -31,16 +32,16 @@
 
 /datum/antagonist/traitor/AI
 	should_specialise = FALSE
+
 /datum/antagonist/traitor/AI/custom
 	silent = TRUE
 	should_give_codewords = FALSE
 	give_objectives = FALSE
 
 
 /datum/antagonist/traitor/on_body_transfer(mob/living/old_body, mob/living/new_body)
-	if(issilicon(new_body) && issilicon(old_body))
-		..()
-	else
+	// human <-> silicon only
+	if(old_body && issilicon(new_body) ^ issilicon(old_body))
```

Comment:
Huh, this looked like it would need another set of parenthesis,

But I looked it up, and ^ has priority over &&.

Wierd, I thought they had same priority.